iterator_methods.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    length.times do |i|
+      yield self[i]
+    end
+  end
+
+  def my_each_with_index
+    length.times do |i|
+      yield self[i], i
+    end
+  end
+
+  def my_select
+    aux_array = []
+    length.times do |i|
+      aux_array << self[i] if yield self[i]
+    end
+    aux_array
+  end
+
+  def my_all?
+    length.times do |i|
+      return false unless yield self[i]
+    end
+    true
+  end
+
+  def my_any?
+    length.times do |i|
+      return true if yield self[i]
+    end
+    false
+  end
+
+  def my_none?
+    length.times do |i|
+      return false if yield self[i]
+    end
+    true
+  end
+
+  def my_count
+    count = 0
+    length.times do |i|
+      count += 1 if yield self[i]
+    end
+    count
+  end
```

Comment:
Not working

```
> ary = [1, 2, 4, 2]
 => [1, 2, 4, 2] 
> ary.count
 => 4 
> ary.my_count
Traceback (most recent call last):
        5: from /usr/local/rvm/rubies/ruby-2.5.1/bin/irb:11:in `<main>'
        4: from (irb):10
        3: from iterator_methods.rb:47:in `my_count'
        2: from iterator_methods.rb:47:in `times'
        1: from iterator_methods.rb:48:in `block in my_count'
LocalJumpError (no block given (yield))
> ary.count(2)
 => 2 
> ary.my_count(2)
Traceback (most recent call last):
        3: from /usr/local/rvm/rubies/ruby-2.5.1/bin/irb:11:in `<main>'
        2: from (irb):12
        1: from iterator_methods.rb:45:in `my_count'
ArgumentError (wrong number of arguments (given 1, expected 0))
```

Reply:
method my_count with arguments.. it's done

Reply:
:+1: